### PR TITLE
cleanup some codes

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: guoyao <1015105054@qq.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
do some minor cleanup
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -1,11 +1,10 @@
 package main
 
 import (
-	_ "sigs.k8s.io/controller-runtime/pkg/metrics"
```
Comment (Garrybest) [inline cmd/agent/main.go:4]: Don't modify this line. If the `sigs.k8s.io/controller-runtime/pkg/metrics` pkg hasn't initialized before `k8s.io/apiserver/pkg/server`, the metrics will be not registed to controller-runtime. See #831

File: cmd/scheduler-estimator/main.go [hunk 2]
```diff
@@ -4,10 +4,9 @@ import (
 	"fmt"
 	"os"
 
-	_ "sigs.k8s.io/controller-runtime/pkg/metrics"
```
Comment (Garrybest) [inline cmd/scheduler-estimator/main.go:7]: +1

File: pkg/estimator/server/server.go [hunk 3]
```diff
@@ -137,6 +137,7 @@ func (es *AccurateSchedulerEstimatorServer) MaxAvailableReplicas(ctx context.Con
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
 		klog.Warningf("No metadata from context.")
+		return nil, fmt.Errorf("no metadata from context")
```
Comment (Garrybest) [inline pkg/estimator/server/server.go:140]: It's just a warning, do not return error.

Comment (gy95) [reply pkg/estimator/server/server.go:140]: if !ok, md will be nil and the following calling `md.Get` will panic?

File: cmd/agent/main.go (hunk 1 above)
Comment (gy95) [reply cmd/agent/main.go]: done.

File: pkg/estimator/server/server.go (hunk 3 above)
Comment (gy95) [reply pkg/estimator/server/server.go]: done.

**Review by RainbowMango — COMMENTED** [review]
I see @Garrybest's comments are all addressed.
/lgtm
/approve

**Pull request merged**